### PR TITLE
Add flag to disable transaction dialogs to web3

### PR DESCRIPTION
**Pull request by Sajma27**
# Summary
A description of the changes proposed in the pull request.

Fixes #number

## Checklist
- [ ] Change is small and easy to review (please split big changes into multiple PRs)
- [ ] Change is consistent with architecture
- [ ] Change is consistent with test architecture
- [ ] Change is consistent with naming conventions
- [ ] New code is covered with tests
- [ ] Tests related to old code are updated
- [ ] Documentation is up to date with changes

### CODE REVIEW

File: universal-login-provider/src/ULIFrameProvider.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ import {Network} from './models/network';
 import {Provider} from './models/provider';
 
 export interface ExtendedConfig extends ProviderConfig {
+  showTransactionDialogs: boolean;
```
Comment (JustynaBroniszewska) [inline universal-login-provider/src/ULIFrameProvider.ts:12]: I would rename it to `showWaitingForTransaction`

File: universal-login-web3/src/ULWeb3Provider.ts [hunk 2]
```diff
@@ -51,7 +52,7 @@ export class ULWeb3Provider implements Provider {
     uiInitializer = initUi,
     observedTokensAddresses,
     browserChecker = new BrowserChecker(),
-  }: ULWeb3ProviderOptions) {
+  }: ULWeb3ProviderOptions, showTransactionDialogs: boolean) {
```
Comment (JustynaBroniszewska) [inline universal-login-web3/src/ULWeb3Provider.ts:55]: Can we make it one of `ULWeb3ProviderOptions` option?

File: universal-login-web3/src/ULWeb3Provider.ts [hunk 3]
```diff
@@ -68,7 +69,8 @@ export class ULWeb3Provider implements Provider {
     this.browserChecker = browserChecker;
     this.walletService = new WalletService(this.sdk, walletFromBrain, sdkConfig.storageService);
 
-    this.uiController = new UIController(this.walletService);
+    this.showTransactionDialogs = showTransactionDialogs;
```
Comment (JustynaBroniszewska) [inline universal-login-web3/src/ULWeb3Provider.ts:72]: We don't need it to be part of ULWeb3Provider

File: universal-login-web3/src/services/ProviderOnlyIframeInitializer.ts [hunk 4]
```diff
@@ -9,9 +9,9 @@ import {SdkConfig} from '@unilogin/sdk';
 export class ProviderOnlyIframeInitializer extends IframeInitializerBase {
   private readonly provider: ULWeb3Provider;
 
-  constructor(endpoint: IframeBridgeEndpoint, network: Network, sdkConfig?: Partial<SdkConfig>) {
+  constructor(endpoint: IframeBridgeEndpoint, network: Network, showTransactionDialogs: boolean, sdkConfig?: Partial<SdkConfig>) {
```
Comment (JustynaBroniszewska) [inline universal-login-web3/src/services/ProviderOnlyIframeInitializer.ts:12]: Let's make sdkConfig and this flag an object: `overrides` 
```ts
constructor(endpoint: IframeBridgeEndpoint, network: Network, overrides: {showTransactionDialogs: boolean, sdkConfig?: Partial<SdkConfig>}) {
}
```

File: universal-login-web3/src/services/setupStrategies.ts [hunk 5]
```diff
@@ -14,7 +14,7 @@ export interface SetupUniLoginOverrides {
   browserChecker?: BrowserChecker;
 }
 
-export const setupUniLogin = (provider: Provider, overrides?: SetupUniLoginOverrides) => ({
+export const setupUniLogin = (provider: Provider, showTransactionDialogs: boolean, overrides?: SetupUniLoginOverrides) => ({
```
Comment (JustynaBroniszewska) [inline universal-login-web3/src/services/setupStrategies.ts:17]: Move it to overrides

File: universal-login-web3/src/services/setupStrategies.ts [hunk 6]
```diff
@@ -24,17 +24,17 @@ export const setupUniLogin = (provider: Provider, overrides?: SetupUniLoginOverr
...
       browserChecker: overrides?.browserChecker,
-    });
+    }, showTransactionDialogs);
   },
 });
 
 export type Strategy = 'UniLogin' | 'Metamask' | Web3ProviderFactory;
 
-export const setupStrategies = (provider: Provider, strategies: Strategy[], overrides?: SetupUniLoginOverrides) => {
+export const setupStrategies = (provider: Provider, strategies: Strategy[], showTransactionDialogs: boolean, overrides?: SetupUniLoginOverrides) => {
```
Comment (JustynaBroniszewska) [inline universal-login-web3/src/services/setupStrategies.ts:33]: Move to overrides

File: universal-login-web3/test/ULWeb3Provider.unit.test.ts [hunk 7]
```diff
@@ -22,7 +22,7 @@ describe('UNIT: ULWeb3Provider', () => {
         relayerUrl: 'http://relayer.url',
         ensDomains: ['uniweb3.eth'],
         observedTokensAddresses: [ETHER_NATIVE_TOKEN.address],
-      });
+      }, true);
```
Comment (JustynaBroniszewska) [inline universal-login-web3/test/ULWeb3Provider.unit.test.ts:25]: We should make it set as default to `true`

File: universal-login-provider/src/ULIFrameProvider.ts (hunk 1 above)
Comment (xitronix) [reply universal-login-provider/src/ULIFrameProvider.ts]: Move it into ProviderConfig

File: universal-login-provider/src/ULIFrameProvider.ts [hunk 8]
```diff
@@ -106,6 +106,7 @@ export class ULIFrameProvider {
 
   static create(network: Network, sdkConfig?: Record<string, any>, config = DEFAULT_CONFIG) {
     return ULIFrameProvider.getInstance({
+      showWaitingForTransaction: config.showWaitingForTransaction,
```
Comment (xitronix) [inline universal-login-provider/src/ULIFrameProvider.ts:109]: it is redundant

File: universal-login-provider/src/ULIFrameProvider.ts [hunk 9]
```diff
@@ -115,6 +116,7 @@ export class ULIFrameProvider {
 
   static createPicker(upstream: Provider | Network, sdkConfig?: Record<string, any>, config = DEFAULT_CONFIG) {
     return ULIFrameProvider.getInstance({
+      showWaitingForTransaction: config.showWaitingForTransaction,
```
Comment (xitronix) [inline universal-login-provider/src/ULIFrameProvider.ts:119]: Unecessary

File: universal-login-provider/src/ULIFrameProvider.ts [hunk 10]
```diff
@@ -106,6 +106,7 @@ export class ULIFrameProvider {
 
   static create(network: Network, sdkConfig?: Record<string, any>, config = DEFAULT_CONFIG) {
```
Comment (xitronix) [inline universal-login-provider/src/ULIFrameProvider.ts:107]: I think we should change config for configOverrides and merge it with DEFAULT_CONFIG then

File: universal-login-web3/src/ULWeb3Provider.ts [hunk 11]
```diff
@@ -13,18 +13,19 @@ import {renderLogoButton} from './ui/logoButton';
...
 export interface ULWeb3ProviderOptions extends Config {
+  showWaitingForTransaction?: boolean;
   sdkConfigOverrides?: Partial<SdkConfig>;
   uiInitializer?: (services: ULWeb3RootProps) => void;
   browserChecker?: BrowserChecker;
 }
 
 export class ULWeb3Provider implements Provider {
-  static getDefaultProvider(networkOrConfig: Network | Config, sdkConfigOverrides?: Partial<SdkConfig>) {
+  static getDefaultProvider(networkOrConfig: Network | Config, overrides: any) {
```
Comment (xitronix) [inline universal-login-web3/src/ULWeb3Provider.ts:23]: any?

File: universal-login-provider/src/buildIframeUrl.ts [hunk 12]
```diff
@@ -1,17 +1,16 @@
 import {Network} from './models/network';
 
 export function buildIframeUrl(
-  iframeUrl: string,
-  picker: boolean,
   sdkConfig: Record<string, any>,
+  configOverrides?: any,
```
Comment (xitronix) [inline universal-login-provider/src/buildIframeUrl.ts:5]: any

File: universal-login-web3/src/ULWeb3Provider.ts [hunk 13]
```diff
@@ -13,18 +13,20 @@ import {renderLogoButton} from './ui/logoButton';
...
 export class ULWeb3Provider implements Provider {
-  static getDefaultProvider(networkOrConfig: Network | Config, sdkConfigOverrides?: Partial<SdkConfig>) {
+  static getDefaultProvider(networkOrConfig: Network | Config, showWaitingForTransaction?: boolean, sdkConfigOverrides?: Partial<SdkConfig>) {
     const config = typeof networkOrConfig === 'string' ? getConfigForNetwork(networkOrConfig) : networkOrConfig;
 
     return new ULWeb3Provider({
+      showWaitingForTransaction,
       ...config,
-      sdkConfigOverrides,
+      ...sdkConfigOverrides,
```
Comment (xitronix) [inline universal-login-web3/src/ULWeb3Provider.ts:29]: why do you deconstruct sdkConfigOverrides

File: universal-login-web3/src/services/ProviderOnlyIframeInitializer.ts [hunk 14]
```diff
@@ -9,9 +9,13 @@ import {SdkConfig} from '@unilogin/sdk';
 export class ProviderOnlyIframeInitializer extends IframeInitializerBase {
   private readonly provider: ULWeb3Provider;
 
-  constructor(endpoint: IframeBridgeEndpoint, network: Network, sdkConfig?: Partial<SdkConfig>) {
+  constructor(
+    endpoint: IframeBridgeEndpoint,
+    network: Network,
+    overrides: {sdkConfig?: Partial<SdkConfig>, showWaitingForTransaction?: boolean},
```
Comment (xitronix) [inline universal-login-web3/src/services/ProviderOnlyIframeInitializer.ts:15]: * overrides should be optional
* extract {sdkConfig?: Partial<SdkConfig>, showWaitingForTransaction?: boolean} to

**Pull request closed without merging**